packages/binding-http/src/http-server.ts

```diff
@@ -208,17 +208,32 @@ export default class HttpServer implements ProtocolServer {
 
     private updateInteractionNameWithUriVariablePattern(
         interactionName: string,
-        uriVariables: PropertyElement["uriVariables"]
+        uriVariables: PropertyElement["uriVariables"],
+        thingVariables: PropertyElement["uriVariables"]
     ): string {
-        if (uriVariables && Object.keys(uriVariables).length > 0) {
+        if (
+            (uriVariables && Object.keys(uriVariables).length > 0) ||
+            (thingVariables && Object.keys(thingVariables).length > 0)
+        ) {
```

Comment:
```suggestion
      const variables = Object.assign({},uriVariables, thingVariables)
        if ( Object.keys(variables).length > 0 ) {
```

Reply:
I am fine with the proposed changes even though I have to say that it seems a matter of taste to me what "simplify" means in this case 🙈.

Moreover, the proposed change create a new variable `variables` which after the `if` is no longer used.
@relu91 If you want me to make these changes I am fine. If you agree this is more a preference change I would go ahead as is...

Reply:
Ah right, I see your point. My main goal was to suggest a shorter and more readable version of the same algorithm (it can be valued objectively by counting lines, code branches, and replicated code). Although complaining about variable creation overhead right now feels like an early optimization problem to me, I leave you to decide 😃 you can merge the PR without addressing my review comments.

Reply:
👍